source/includes/connection-snippets/x509-snippets/java-connection.java

```diff
@@ -0,0 +1,25 @@
+
+// begin x509 connection
+import java.util.Arrays;
+import org.bson.Document;
+import com.mongodb.MongoClientSettings;
+import com.mongodb.MongoCredential;
+import com.mongodb.ServerAddress;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoIterable;
+
+
+System.setProperty("javax.net.ssl.trustStore","/etc/certs/mongodb/client.truststore");
+System.setProperty("javax.net.ssl.trustStorePassword","mypass");
+
+System.setProperty("javax.net.ssl.keyStore", "/etc/certs/mongodb/client.keystore");
+System.setProperty("javax.net.ssl.keyStorePassword", "mypass");
+String subject = "<your_subject>"; // CN=ChrisChoClient,OU=TestClientCertificateOrgUnit,O=TestClientCertificateOrg,L=TestClientCertificateLocality,ST=TestClientCertificateStateC=US
+MongoCredential credential = MongoCredential.createMongoX509Credential(subject); 
+MongoClientSettings settings = MongoClientSettings.builder()
+        .credential(credential)
+        .applyToClusterSettings(builder -> builder.hosts(Arrays.asList(new ServerAddress("localhost"))))
+        .applyToSslSettings(builder -> builder.enabled(true)).build();
+MongoClient client = MongoClients.create(settings);
+// end x509 connection
```

Comment:
I feel like we should at least put these things in a file that can be run -- this won't run at all because there isn't a main method and it isn't valid java, for instance. Because of the way async methods work in java (and C# and C++, I believe) it's a little weird to omit those details too.

Reply:
I agree that it is a little weird, the password auth implementations currently on the driver docs don't have wrapper classes for the most part (it is a little in-consistent though). See:
https://docs.mongodb.com/ecosystem/drivers/cxx/

@terakilobyte suggested have these only having the client connection code without main methods or classes to wrap the code.

Reply:
Okay, returned back to the original state where there is a main method. The new code is basically the same as chris's original x509 repo. Resolving this discussion.